src/transformers/models/omdet_turbo/modeling_omdet_turbo.py

```diff
@@ -0,0 +1,2034 @@
+            `(batch_size, sequence_length, hidden_size)`. Hidden-states of the model at the output of each layer
+            plus the initial embedding outputs.
+        encoder_attentions (`Optional[Tuple[Tuple[torch.FloatTensor]]]`):
+            Tuple of tuples of `torch.FloatTensor` (one for attention for each layer) of shape `(batch_size, num_heads,
+            sequence_length, sequence_length)`. Attentions weights after the attention softmax, used to compute the
+            weighted average in the self-attention, cross-attention and multi-scale deformable attention heads.
+        enc_outputs_coord_logits (`torch.FloatTensor` of shape `(batch_size, num_queries, 4)`):
+            The predicted coordinates of the objects from the encoder.
+        enc_outputs_class_logits (`Tuple[torch.FloatTensor]`):
+            The predicted class of the objects from the encoder.
+        encoder_last_hidden_state (`Optional[torch.FloatTensor]`):
+            Last layer hidden-state of the encoder.
+    """
+
+    decoder_coord_logits: torch.FloatTensor = None
+    decoder_class_logits: torch.FloatTensor = None
+    vision_model_outputs: Optional[Tuple[torch.FloatTensor]] = None
+    text_model_outputs: Optional[Tuple[torch.FloatTensor]] = None
+    init_reference_points: torch.FloatTensor = None
+    intermediate_reference_points: Optional[Tuple[Tuple[torch.FloatTensor]]] = None
+    decoder_hidden_states: Optional[Tuple[torch.FloatTensor]] = None
+    decoder_attentions: Optional[Tuple[Tuple[torch.FloatTensor]]] = None
+    encoder_hidden_states: Optional[Tuple[torch.FloatTensor]] = None
+    encoder_attentions: Optional[Tuple[Tuple[torch.FloatTensor]]] = None
+    enc_outputs_coord_logits: torch.FloatTensor = None
+    enc_outputs_class_logits: Optional[Tuple[torch.FloatTensor]] = None
+    encoder_last_hidden_state: Optional[torch.FloatTensor] = None
+
+
+@add_start_docstrings(
+    """
+    OmDetTurbo Model (consisting of a vision and a text backbone, and encoder-decoder architecture) outputting
+    bounding boxes and classes scores for tasks such as COCO detection.
+    """,
+    OMDET_TURBO_START_DOCSTRING,
+)
+class OmDetTurboForObjectDetection(OmDetTurboPreTrainedModel):
+    def __init__(self, config: OmDetTurboConfig):
+        super().__init__(config)
+        # Create backbone
+        self.backbone = OmDetTurboVisionBackbone(config)
+        self.language_backbone = OmDetTurboLanguageBackbone(config=config)
+        self.encoder = OmDetTurboHybridEncoder(config)
+        self.decoder = OmDetTurboDecoder(config)
+        self.num_queries = config.num_queries
+
+        self.language_cache_label = OmDetTurboLRUCache(config.cache_size)
+        self.language_cache_prompt = OmDetTurboLRUCache(config.cache_size)
+        self.post_init()
+
+    @add_start_docstrings_to_model_forward(OMDET_TURBO_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=OmDetTurboModelOutput, config_class=_CONFIG_FOR_DOC)
+    def forward(
```

Comment:
Ended up removing OmDetTurboModel altogether as discussed

Reply:
Sorry - I missed this as it was marked as resolved. 

> but all models should have FooModel

Generally, yes, we want this. It's not a hard-and-fast rule though, other models like Fuyu have just a task-specific model. 

However, yes, you're right in this case - as there's decoder heads we can and should separate this out in order to have an OmDetTurboModel

Reply:
Added now!

Reply:
Oh I see, makes sense. I guess I don't have a choice but to keep the encoder_bbox_head and encoder_class_head in the decoder though as they are used early on in the forward pass no?

Reply:
Looking further into it, I'm not sure I can isolate `decoder_class_head` and `decoder_bbox_head` in `OmDetTurboForObjectDetection`, as they are used in the main decoder loop to refine bboxes and classes iteratively. I would need to move the whole decoder to `OmDetTurboForObjectDetection`, but I don't know if it would make sense to then have `OmDetTurboModel` just being the encoder. Wdyt @amyeroberts @qubvel ?

Reply:
OK, makes sense. In this case, I'd say we just have the encoder, decoder and a OmDetTurboForObjectDetection class which contains the encoder and decoder